SlidePuzzle/src/main/java/ThirdApproach/Puzzle.java

```diff
@@ -0,0 +1,59 @@
+package ThirdApproach;
+
+public class Puzzle {
```

Comment:
Looks like all the methods in this class are static, so...  Should it static class?